runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -10412,227 +10412,769 @@ static TR::Register *inlineIsAssignableFrom(TR::Node *node, TR::CodeGenerator *c
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwinm, node, endReg, endReg, 1,
+                                         0xFFFFFFFFFFFFFFFE);
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwinm, node, vendReg, vendReg, 1,
+                                         0xFFFFFFFFFFFFFFFE);
+         break;
+      case TR::Int32:
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwinm, node, endReg, endReg, 2,
+                                         0xFFFFFFFFFFFFFFFC);
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwinm, node, vendReg, vendReg, 2,
+                                         0xFFFFFFFFFFFFFFFC);
+         break;
+      default:
+         TR_ASSERT_FATAL(false, "Unsupported hashCodeHelper elementType");
+      }
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, valueReg, valueReg, endReg);
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, endReg, valueReg, vendReg);
+
+   // initialisations
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::XOR, node, tempReg, tempReg, tempReg);
+   loadConstant(cg, node, 0x0, constant0Reg);
+
+   // using the serial loop is faster if there are less than 16 items
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::cmpi4, node, condReg, vendReg, 16*elementSize);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::blt, node, serialLabel, condReg);
+
+   // load multiplier (anything with more than 4 bytes can be truncated)
+   // 31^0-6 = 1 31 961 29791 923521 28629151 887503681
+   // 31^7 = 667E12CDF           31^8 = C694446F01         31^9 = 180BF449711F 31^10 = 2E97294E4B2C1
+   // 31^11 = 5A44E007B1A55F     31^12 = AEE5720EE830681     31^13 = 152DC8CFCE1DDC99F
+   // 31^14 = 2908B5129F59DB6A41 31^15 = 4F80DED414BE191DDDF 31^16 = 9A09AFBAE83050A9DE01
+   static uint32_t multiplierVectors_be8[24] = {0x50A9DE01, 0x50A9DE01, 0x50A9DE01, 0x50A9DE01,
+      0xE191DDDF, 0x59DB6A41, 0xE1DDC99F, 0xEE830681, 0x07B1A55F, 0x94E4B2C1, 0xF449711F, 0x94446F01,
+      0x67E12CDF, 887503681, 28629151, 923521, 29791, 961, 31, 1, 0, 0, 0, 0};
+   static uint32_t multiplierVectors_le8[24] = {0x50A9DE01, 0x50A9DE01, 0x50A9DE01, 0x50A9DE01,
+      1, 31, 961, 29791, 923521, 28629151, 887503681, 0x67E12CDF,
+      0x94446F01, 0xF449711F, 0x94E4B2C1, 0x07B1A55F, 0xEE830681, 0xE1DDC99F, 0x59DB6A41, 0xE191DDDF,
+      0, 0, 0, 0};
+   static uint32_t multiplierVectors_be16[16] = {0x94446F01, 0x94446F01, 0x94446F01, 0x94446F01,
+      0x67E12CDF, 887503681, 28629151, 923521, 29791, 961, 31, 1, 0, 0, 0, 0};
+   static uint32_t multiplierVectors_le16[16] = {0x94446F01, 0x94446F01, 0x94446F01, 0x94446F01,
+      1, 31, 961, 29791, 923521, 28629151, 887503681, 0x67E12CDF, 0, 0, 0, 0};
+   static uint32_t multiplierVectors_be32[12] = {923521, 923521, 923521, 923521,
+                                                29791, 961, 31, 1, 0, 0, 0, 0};
+   static uint32_t multiplierVectors_le32[12] = {923521, 923521, 923521, 923521,
```

Comment:
this made me uneasy at least.  these constants are set up as C/C++ static data in libj9jit29.so module. what happen if that module is unloaded? better put them in PersistentMemory like pseudo TOC. Also, this must be the reason it is disabled for AOT and JITServer.